src/Raven.Studio/typescript/components/pages/resources/manageServer/storageReport/partials/StorageReportTable.tsx

```diff
@@ -0,0 +1,101 @@
+import { hasChildren, StorageReportItem } from "components/pages/resources/manageServer/storageReport/partials/common";
+import genUtils from "common/generalUtils";
+import React, { useId } from "react";
+import { withPreventDefault } from "components/utils/common";
+import { UncontrolledTooltip } from "reactstrap";
+import useUniqueId from "hooks/useUniqueId";
+
+interface StorageReportTableProps {
+    node: StorageReportItem;
+    root: StorageReportItem;
+    onNodeSelected: (node: StorageReportItem) => void;
+}
+
+export function StorageReportTable(props: StorageReportTableProps) {
+    const { node, root, onNodeSelected } = props;
+    const tooltipId = useUniqueId("tooltip");
+
+    if (!node || !node.internalChildren) {
+        return null;
+    }
+
+    const showPagesColumn = node.internalChildren.some((x) => x.type === "tree");
+    const showEntriesColumn = node.internalChildren.some((x) => x.type === "table" || x.type === "tree");
+    const showTempFiles = node === root;
```

Comment:
We use the prefixes 'is', 'has' or 'can' for boolean values. This makes it easier to read such code when we do not have an IDE, e.g. on review. I suggest changing the name to 'isPagesColumnVisible' and similarly for the rest. The 'show' prefis suggests that this is a function that shows something.